src/System/IO/Streams/SequenceId.hs

```diff
@@ -49,26 +55,39 @@ sequenceIdInputStream initSeqId getSeqId seqIdFaultHandler inStream =
 -- Example:
 --
 -- @
--- (outStream', getSeqId) <- 'sequenceIdOutputStream' 1 outStream
--- return $ 'System.IO.Streams.Combinators.contramapM' (addSeqId getSeqId) outStream'
+-- ghci> import qualified System.IO.Streams as Streams
+-- ghci> (os, getList) <- Streams.listOutputStream :: IO (OutputStream (Int,Int), IO [(Int,Int)])
+-- ghci> (outStream', getSeqId) <- sequenceIdOutputStream 0 (\seqId a -> (seqId, a)) os
+-- ghci> Streams.write (Just 6) outStream'
+-- ghci> Streams.write (Just 7) outStream'
+-- ghci> getList
+-- [(1,6),(2,7)]
+-- ghci> getSeqId
+-- 2
+-- ghci> Streams.write (Just 6) outStream'
+-- ghci> Streams.write (Just 7) outStream'
+-- ghci> getList
+-- [(1,6),(2,7)]
 -- @
 sequenceIdOutputStream :: Integral s
-                       => s                   -- ^ Initial sequence ID
-                       -> (s -> a -> b)       -- ^ Transformation function
-                       -> OutputStream b      -- ^ 'System.IO.Streams.OutputStream' to count the elements of
-                       -> IO (OutputStream a) -- ^ ('IO' 'SequenceId') is the action to run to get the current sequence ID
+                       => s                         -- ^ Initial sequence ID
+                       -> (s -> a -> b)             -- ^ Transformation function
+                       -> OutputStream b            -- ^ 'System.IO.Streams.OutputStream' to count the elements of
+                       -> IO (OutputStream a, IO s) -- ^ returns a new stream as well as an IO action to fetch and reset the updated seed value
 sequenceIdOutputStream i f = outputFoldM f' i
   where f' seqId bdy = (nextSeqId, f nextSeqId bdy)
           where nextSeqId = incrementSeqId seqId
 
 
-outputFoldM :: (a -> b -> (a, c))  -- ^ fold function
-            -> a                   -- ^ initial seed
-            -> OutputStream c      -- ^ output stream
-            -> IO (OutputStream b) -- ^ returns a new stream as well as an IO action to fetch the updated seed value.
+outputFoldM :: Integral a
+            => (a -> b -> (a, c))        -- ^ fold function
+            -> a                         -- ^ initial seed
+            -> OutputStream c            -- ^ output stream
+            -> IO (OutputStream b, IO a) -- ^ returns a new stream as well as an IO action to fetch and reset the updated seed value.
 outputFoldM step initSeqId outStream = do
```

Comment:
It looks like `outputFoldM` is a part of io-streams now.

https://hackage.haskell.org/package/io-streams-1.5.2.1/docs/System-IO-Streams-Combinators.html#v:outputFoldM

Maybe you could use that? I'm not sure.

Reply:
this function is extremely similar to `outputFoldM`, but the difference is the fold function because in our version we are doing a transformation as well. So `outputFoldM` has a fold function of `(a -> b -> IO a)`, but ours is `(a -> b -> (a, c))`.  The code is almost identical to `outputFoldM` in io-streams.